h5netcdf/core.py

```diff
@@ -132,21 +132,19 @@ def shuffle(self):
 
 NOT_A_VARIABLE = b'This is a netCDF dimension but not a netCDF variable.'
 
-class lazy_objects(Mapping):
-    def __init__(self, object_cls):
-        self._objects = list()
-        self._loaded_objects = dict()
+class _object_names(Mapping):
```

Comment:
Use `CapWords` for class names per PEP 8. Maybe `_LazyObjectLookup` would be appropriate here?